### DISCUSSION TITLE
How to authenticate users using unsupported oauth provider ?

### DISCUSSION THREAD

**Discussion by leElvyn** [pocketbase/pocketbase#840] [Q&A]
Hi, i need to authenticate users using a microsoft oauth2. Currently, i cannot find how to authenticate a user with only his email. What can i do ?

**Comment by ganigeorgiev** [comment]
Microsoft OAuth2 currently is not supported (you could check this issue - https://github.com/pocketbase/pocketbase/issues/746).

>  i cannot find how to authenticate a user with only his email

If you are using the JavaScript SDK, you can call `client.users.authViaEmail("test@example.com", "123456")`.
For admins authentication - `client.admins.authViaEmail("test@example.com", "123456")`.

**Reply by leElvyn** [reply]
Via email only*
To authenticate in the backend, using an admin account a user after i validated via oauth that that email belongs that session

**Reply by ganigeorgiev** [reply]
Currently this can be handled only if you use PocketBase as framework and register your user in the request context. You can see an example with custom middleware that reads the user email from a trusted header in https://github.com/pocketbase/pocketbase/issues/136#issuecomment-1186191919.

In the future there will be support to generate user tokens by admins. You could subscribe to this issue - https://github.com/pocketbase/pocketbase/issues/718.
